x-pack/plugins/case/server/client/cases/update.ts

```diff
@@ -10,36 +10,331 @@ import { pipe } from 'fp-ts/lib/pipeable';
+  };
+
+  const requestsUpdatingTypeField = requests.filter((req) => req.type === CaseType.collection);
+  const casesAlertTotals = await Promise.all(
+    requestsUpdatingTypeField.map((caseToUpdate) => getAlertsForID(caseToUpdate))
+  );
+
+  // grab the cases that have at least one alert comment attached to them
+  const typeUpdateWithAlerts = casesAlertTotals.filter((caseInfo) => caseInfo.alerts.total > 0);
+
+  if (typeUpdateWithAlerts.length > 0) {
+    const ids = typeUpdateWithAlerts.map((req) => req.id);
+    throw Boom.badRequest(
+      `Converting a case to a collection is not allowed when it has alert comments, ids: [${ids.join(
+        ', '
+      )}]`
+    );
+  }
+}
+
+/**
+ * Get the id from a reference in a comment for a specific type.
+ */
+function getID(
+  comment: SavedObject<CommentAttributes>,
+  type: typeof CASE_SAVED_OBJECT | typeof SUB_CASE_SAVED_OBJECT
+): string | undefined {
+  return comment.references.find((ref) => ref.type === type)?.id;
+}
+
+/**
+ * Gets all the alert comments (generated or user alerts) for the requested cases.
+ */
+async function getAlertComments({
+  casesToSync,
+  caseService,
+  client,
+}: {
+  casesToSync: ESCasePatchRequest[];
+  caseService: CaseServiceSetup;
+  client: SavedObjectsClientContract;
+}): Promise<SavedObjectsFindResponse<CommentAttributes>> {
+  const idsOfCasesToSync = casesToSync.map((casePatchReq) => casePatchReq.id);
+
+  // getAllCaseComments will by default get all the comments, unless page or perPage fields are set
+  return caseService.getAllCaseComments({
+    client,
+    id: idsOfCasesToSync,
+    includeSubCaseComments: true,
+    options: {
+      filter: `${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.alert} OR ${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.generatedAlert}`,
+    },
+  });
+}
+
+/**
+ * Returns a map of sub case IDs to their status. This uses a group of alert comments to determine which sub cases should
+ * be retrieved. This is based on whether the comment is associated to a sub case.
+ */
+async function getSubCasesToStatus({
+  totalAlerts,
+  caseService,
+  client,
+}: {
+  totalAlerts: SavedObjectsFindResponse<CommentAttributes>;
+  caseService: CaseServiceSetup;
+  client: SavedObjectsClientContract;
+}): Promise<Map<string, CaseStatuses>> {
+  const subCasesToRetrieve = totalAlerts.saved_objects.reduce((acc, alertComment) => {
+    if (
+      isCommentRequestTypeAlertOrGenAlert(alertComment.attributes) &&
+      alertComment.attributes.associationType === AssociationType.subCase
+    ) {
+      const id = getID(alertComment, SUB_CASE_SAVED_OBJECT);
+      if (id !== undefined) {
+        acc.add(id);
+      }
+    }
+    return acc;
+  }, new Set<string>());
+
+  const subCases = await caseService.getSubCases({
+    ids: Array.from(subCasesToRetrieve.values()),
+    client,
+  });
+
+  return subCases.saved_objects.reduce((acc, subCase) => {
+    // log about the sub cases that we couldn't find
```

Comment:
Is this a `Todo` to notify the user about situations where there is an error?

Reply:
Yeah it's really a TODO to add logging later